src/backend/update_traversability_functor.cpp

```diff
@@ -0,0 +1,493 @@
+  const auto view = info->loop_closure_detected ? LayerView(places)
+                                                : active_tracker_.view(places, true);
+
+  // Compare all (newly) archived nodes against inactive candidate nodes.
+  MergeList result;
+  std::set<NodeId> merged;
+  for (const auto& from_node : view) {
+    if (merged.count(from_node.id)) {
+      continue;
+    }
+
+    auto& from_attrs = from_node.attributes<TraversabilityNodeAttributes>();
+    if (from_attrs.is_active) {
+      continue;
+    }
+
+    auto from_boundary = Boundary(from_attrs);
+    for (const auto to_id : nn_->findRadius(from_attrs.position, radius_, false)) {
+      if (to_id == from_node.id || merged.count(to_id)) {
+        continue;
+      }
+      auto& to_attrs = dsg.getNode(to_id).attributes<TraversabilityNodeAttributes>();
+      // Avoid merging nodes with active window (temporal) overlap.
+      if (from_attrs.last_observed_ns >= to_attrs.first_observed_ns &&
+          from_attrs.first_observed_ns <= to_attrs.last_observed_ns) {
+        continue;
+      }
+
+      // Check boundaries.
+      auto to_boundary = Boundary(to_attrs);
+      if (!to_boundary.intersects(from_boundary)) {
+        continue;
+      }
+
+      // Simple case: If places are completely contained.
+      if (isContained(to_boundary, from_boundary)) {
+        result.push_back({to_id, from_node.id});
+        merged.insert(to_id);
+        continue;
+      }
+      if (isContained(from_boundary, to_boundary)) {
+        result.push_back({from_node.id, to_id});
+        merged.insert(from_node.id);
+        continue;
+      }
+
+      // Large overlap: No need to keep both, as transition is always possible between.
+      auto intersection = from_boundary.intersection(to_boundary);
+      if (intersection.height() < config.min_place_size ||
+          intersection.width() < config.min_place_size) {
+        // TODO(lschmid): Can consider cropping and dropping in the future.
+        // Always update the classification of the boundary sides.
+        overlapping_nodes_to_cleanup_.insert({from_node.id, to_id});
+        continue;
+      }
+
+      // Aligned in X or Y direction: Merge by fusing boundary.
+      if ((std::abs(from_boundary.min.x() - to_boundary.min.x()) < config.tolerance &&
+           std::abs(from_boundary.max.x() - to_boundary.max.x()) < config.tolerance) ||
+          (std::abs(from_boundary.min.y() - to_boundary.min.y()) < config.tolerance &&
+           std::abs(from_boundary.max.y() - to_boundary.max.y()) < config.tolerance)) {
+        merged.insert(to_id);
+        result.push_back({to_id, from_node.id});
+        continue;
+      }
+
+      // If not fuseable easily keep larger.
+      if (from_boundary.area() >= to_boundary.area()) {
+        result.push_back({to_id, from_node.id});
+        merged.insert(to_id);
+      } else {
+        result.push_back({from_node.id, to_id});
+        merged.insert(from_node.id);
+      }
+    }
```

Comment:
(minor) I'd maybe split this up into a helper function or two, but definitely not needed